src/plugins/Wallet/database/Wallet.db.ts

```diff
@@ -86,11 +87,26 @@ export const createWalletDBAccess = createDBAccess(() => {
                     await token.update(token.value)
                 }
             }
+            /**
+             * Fix providerType does not exist in legacy wallet
+             */
+            async function v4_v5() {
+                const t = createTransaction(db, 'readwrite')('Wallet', 'ERC20Token')
+                const wallets = t.objectStore('Wallet')
+                for await (const wallet of wallets) {
+                    const wallet_ = wallet as any
+                    if (wallet_.value.provider) continue
+                    if (wallet_.value.type === 'managed') wallet_.value.provider = ProviderType.Maskbook
```

Comment:
IMO this doesn't need a DB upgrade. You can add the patch on the WalletRecordOutDB